src/plugins/intel_gna/gna_plugin.cpp

```diff
@@ -1128,7 +1140,27 @@ void GNAPlugin::LoadNetwork(const CNNNetwork& _network) {
     } else {
         for (auto& inputLayer : inputLayers) {
             if (LayerInfo(inputLayer).isInput()) {
-                inputs_ptr_->at(inputLayer->name).orientation = kDnnInterleavedOrientation;
+                auto doesntHaveGnaMapping = [this](CNNLayerPtr l) {
+                    auto dnnLayer = graphCompiler.dnnComponents.findComponent(l);
+                    return dnnLayer == nullptr;
+                };
+
+                auto nextLayers = CNNNetGetAllNextLayersSkipCertain(inputLayer, -1, doesntHaveGnaMapping);
+
+                if (nextLayers.size() != 1) {
+                    inputs_ptr_->at(inputLayer->name).orientation = kDnnInterleavedOrientation;
+                    continue;
+                }
+
+                auto component = graphCompiler.dnnComponents.findComponent(nextLayers.front());
+                auto dims = inputs_ptr_->at(inputLayer->name).dims;
+
+                if (component && dims.size() >= 2 && component->num_rows_in == dims[0] &&
+                    component->num_columns_in == dims[1]) {
```

Comment:
we should reuse the same condition as in inputs (eg create a function in the component class ?)